adapters/amqp-vertx/src/main/java/org/eclipse/hono/adapter/amqp/VertxBasedAmqpProtocolAdapter.java

```diff
@@ -286,64 +286,60 @@ private void handleSessionOpen(final ProtonConnection conn, final ProtonSession
     }
 
     /**
-     * This method is invoked when an AMQP Attach frame is received by this server.
+     * This method is invoked when an AMQP Attach frame is received by this server. If the receiver link contains
+     * a target address, this method simply closes the link, otherwise, it accept and open the link.
      * 
      * @param receiver The receiver link for receiving the data.
      * @param conn The connection through which the request is initiated.
      */
     protected void handleRemoteReceiverOpen(final ProtonReceiver receiver, final ProtonConnection conn) {
-        final Device authenticatedDevice = conn.attachments().get(AmqpAdapterConstants.KEY_CLIENT_DEVICE,
-                Device.class);
-        validateEndpoint(receiver, authenticatedDevice).recover(t -> {
-            LOG.debug(
-                    "fail to establish a receiving link with client [{}] due to an invalid endpoint [{}]."
-                            + " closing the link",
-                    receiver.getName(), receiver.getRemoteQoS());
-            receiver.setCondition(AmqpContext.getErrorCondition(t));
+        if (receiver.getRemoteTarget() != null) {
+            if (receiver.getRemoteTarget().getAddress() != null && !receiver.getRemoteTarget().getAddress().isEmpty()) {
+                LOG.debug("Closing link due to the present of Target [address : {}]", receiver.getRemoteTarget().getAddress());
+            }
+            receiver.setCondition(
+                    AmqpContext.getErrorCondition(new ClientErrorException(HttpURLConnection.HTTP_BAD_REQUEST,
+                            "This adapter does not accept a target address on receiver links")));
             receiver.close();
-            return Future.failedFuture(t);
-        }).compose(ok -> {
+        } else {
+            final Device authenticatedDevice = conn.attachments().get(AmqpAdapterConstants.KEY_CLIENT_DEVICE,
+                    Device.class);
             LOG.debug("Established receiver link at [address: {}]",
                     (receiver.getRemoteTarget() != null) ? receiver.getRemoteTarget().getAddress() : null);
 
-            receiver.setTarget(receiver.getRemoteTarget());
+            receiver.setTarget(null);
             receiver.setQoS(receiver.getRemoteQoS());
             if (ProtonQoS.AT_LEAST_ONCE.equals(receiver.getRemoteQoS())) {
                 // disable auto-accept for this transfer model.
                 // in this case, the adapter will apply the required disposition
                 receiver.setAutoAccept(false);
             }
             receiver.handler((delivery, message) -> {
-                uploadMessage(new AmqpContext(delivery, message, receiver.getRemoteTarget(), authenticatedDevice));
+                uploadMessage(receiver.getRemoteQoS(), new AmqpContext(delivery, message, authenticatedDevice));
```

Comment:
the device/gateway will use this link to send telemetry and/or event messages using both *at least once* and *at most once*. The QoS indicated on link establishment therefore is not very useful. Instead, you need to determine the QoS for each message based on whether the message is already settled or not ...